### ISSUE TITLE
BUG: Transaction is not being published

### ISSUE THREAD

**Issue by Dmatafonov** [paritytech/parity#7924]
> I'm running:
>
> - **Which Parity version?**: version Parity/v1.9.2-beta-0feb0bb-
> - **Which operating system?**: Linux
> - **How installed?**: binaries
> - **Are you fully synchronized?**: yes
> - **Which network are you connected to?**: ethereum
> - **Did you try to restart the node?**: no

I create transaction with `personal_sendTransaction`, it outputs me tx_id, but transaction never shows up on the blockchain.

It is really annoying and looks like a HUGE bug, because it is basic feature of any node - send funds from one account to another.

Here is my `l-own_tx` log

```
DEBUG own_tx  Imported to Future (hash d956e137bc52dafbc0eeac5c2ec3a4b5d9563f85879fc6600eb575507ae21504)
```

![image](https://user-images.githubusercontent.com/8863854/36342980-04cfd80c-1417-11e8-8e96-021332cc1c1c.png)

Imported to the future? To what future? To 2020?

Nothing on blockchain
https://etherscan.io/tx/0xd956e137bc52dafbc0eeac5c2ec3a4b5d9563f85879fc6600eb575507ae21504

My previous troubles with the same issue ([transaction is not being published](https://github.com/paritytech/parity/issues/7872)).

_Sometime_ it starts working after node restart. But it isnt a good idea to restart node before each new transaction?

_I send ether to my clients and provide them ouputted tx_id._
_And they ask me then after: why do we dont see incoming funds and transacion on the blockchain?_
_- Ah, that's because parity works this way, let my try again._

**Comment by Dmatafonov** [comment]
And yes, Im fully synced. At the time of writing I have 5107572 blocks, one block ahead of etherscan.io

**Comment by Dmatafonov** [comment]
I put 50GWei for gasPrice and 21000 gas fro transaction. Price doesnt help, something always starts to go wrong and transactions are not sent anymore.

**Comment by 5chdn** [comment]
Please restart with --no-persistent-txqueue and try again; this could be a nonce issue.

**Comment by Dmatafonov** [comment]
> Please restart with --no-persistent-txqueue and try again; this could be a nonce issue.

The node is always started with --no-persistent-txqueue

> this could be a nonce issue.

OK, then what? Is it expected behaivior, not a bug?

**Comment by 5chdn** [comment]
Hold on.

**Comment by 5chdn** [comment]
There must be any transaction stuck before that, can you check? Otherwise it would not import it into the future :)

**Comment by Dmatafonov** [comment]
> There must be any transaction stuck before that, can you check? 

I looked in pending transactions - nothing. What you mean by 'stuck' and where should I look?

**Comment by Dmatafonov** [comment]
Can I get UP on this? Please?

Today had the same issue, received tx_id and sent it to client, but transaction never appeared on the blockchain.

**Comment by Dmatafonov** [comment]
If you cant fix it may be you tell if there is a reliable way to check that something wrong? Lets say transaction SHOULD appear on the blockchain in one hour or so.

Then I'll write the script to check all those stuck transactions, and will restart node and will send again. I dont kbow how others use parity, but Im using it fully automated, so I dont even know id it stops working and need to be restarted.

**Comment by rphmeier** [comment]
@Dmatafonov are you choosing the nonce manually for the transaction? or allowing the node to choose it for you?

**Comment by Dmatafonov** [comment]
@rphmeier, I dont choose it manually. All I do is I call metheod `personal_sendTransaction` and it does return me txid always.

**Comment by rphmeier** [comment]
@Dmatafonov if you allow the node to choose it it shouldn't be importing into the future queue (which is what happens when next usable nonce is X but the transaction has at least X + 1). 

@tomusdrw maybe something with the new RPC nonce reservation thing?

**Comment by Dmatafonov** [comment]
I even tried to resintall everything. I bought new VPS and installed everything from scratch,  hoping that it was come kind local mess up.

And I never had this issue with Geth (though I had too many other issues with geth).

>t it shouldn't be importing into the future queue

I saw in tx-que-log `imported to the future`

**Comment by Dmatafonov** [comment]
@rphmeier @5chdn any update on this?

**Comment by jakecraige** [comment]
I'm seeing a similar issue from time to time. A transaction is broadcast and is shortly marked invalid. Later I attempt to broadcast the same transaction and it broadcasts successfully. The transaction content is the exact same in both.

This is the first transaction from an account that had just received funds. I attempt to broadcast right after it receives them. This is happening constantly (100+ of times an hour) and only see this come up once every few days.

This is what I see in the logs. Data obfuscated for privacy, logs filtered to messages that include the relevant transaction hash. If there's any other info that would be helpful I can provide what I can via email/DMs. If this seems like a separate issue I can open another.

This is on Parity 1.8.10

![kibana](https://user-images.githubusercontent.com/3799709/37438898-8451e42c-27b2-11e8-9772-3deb2f592e33.png)

**Comment by kryvel** [comment]
Hi, I have the same issue.
But only when I send nonce parametr.
Any update on this ? TIA

**Comment by folsen** [comment]
This is a really vague and general issue where the cause could come from many different factors. The best way to debug would get on chat and help us reproduce, however, it may be addressed by the new transaction queue in v1.11.1

@tomusdrw Please reopen if you think this is _not_ addressed at all by tx queue and merits further investigation.

To all others on this thread, I encourage you to find us on https://riot.im/app/#/room/#support:matrix.parity.io
